site/docs/transform/stack.md

```diff
@@ -94,8 +96,47 @@ If you want to define custom sort order, you can derive a new field using the [`
 
 Note: we plan to have [a better syntax for customized sort order](https://github.com/vega/vega-lite/issues/2915) in the future.
 
-## Layering Lines on top of Stacked Area Chart
+### Layering Lines on top of Stacked Area Chart
 
 Since `line` marks are not stacked by default, to layer lines on top of stacked area charts, you have to manually set the `stack` offset for the lines.
 
 <div class="vl-example" data-name="normalized/stacked_area_overlay_normalized"></div>
+
+{:#transform}
+## Stack Transform
+
+{: .suppress-error}
+```json
+// A View Specification
+{
+  ...
+  "transform": [
+    {"stack": ..., "groupby": ..., "offset": ..., "sort": ..., "as" ...} // Stack Transform
+     ...
+  ],
+  ...
+}
+```
+
+The `stack` transform in the `transform` array has the following properties:
+
+{% include table.html props="stack,groupby,offset,sort,as" source="StackTransform" %}
+
+### Diverging Bar Chart
+
+Here we initially `stack` by `"question"` and then use the `window` transform to offset each stack.
+<div class="vl-example" data-name="bar_diverging_stack_transform"></div>
+
+### Basic Mosaic Chart
```

Comment:
1) I don't know if we need to show all of these variant of mosaic chart examples.

Reply:
For 2., I'm asking @domoritz  to add `json-stringify-pretty-compact` to  bit.ly/js2json -- so we can use it.

Reply:
I kept only the simple mosaic example. Will make a separate PR for adding `rect_mosaic_labelled_with_offset`to the gallery and then add a link to it.

Reply:
Let's just do it here.